src/cephadm/cephadm

```diff
@@ -3788,6 +3795,30 @@ def prepare_bootstrap_config(
     cp.set('global', 'fsid', fsid)
     cp.set('global', 'mon host', mon_addr)
     cp.set('global', 'container_image', image)
+
+    if ctx.single_host_defaults:
+        logger.info('Adjusting default settings to suit single-host cluster...')
+        # replicate across osds, not hosts
+        if (
+                not cp.has_option('global', 'osd_crush_choose_leaf_type')
+                and not cp.has_option('global', 'osd crush choose leaf type')
+        ):
+            cp.set('global', 'osd_crush_choose_leaf_type', '0')
```

Comment:
@liewegas I'm guessing type `0` means OSD (because that's what makes sense in this context); so, unless this is meant for a cluster that won't grow beyond a single node, setting OSD from the start will always require the admin to change the crushmap to re-set it to host/whatever, right? AFAIK, there's no straightforward way to readjust the crush choose leaf type any other way.

I'm guessing this is fine, the admin can always readjust, but it might be worth noting that somewhere. I can see this being useful for seed clusters [1], regardless of whether it's meant for a deployment that won't grow beyond a single-node for whatever reason.

[1] This will certainly be useful for initial node deployment in aquarium, but regardless of whether we end up relying on this flag or not, we get the benefit of doing some more logic post-bootstrap, which might not be the case here.

Reply:
Yeah, this is intended for clusters that will remain single-node, not once that will expand later.  If the goal is to get to HEALTH_OK on a single node, I think you're already covered, since we don't create the device_health_metrics pool until we have enough OSDs.  (I think.)  Or you can adjust min_size to 1 and then back up again later.

Reply:
Yeah, just thought I'd check to make sure that was the intended purpose. :)